src/lib/registration/controllers/addons_selection.rb

```diff
@@ -0,0 +1,259 @@
+        @initial_addons_selection = Addon.selected.dup
+        preselect_recommended
+      end
+
+      # Returns a collection of {Item} representing available addons
+      #
+      # @see #item_for
+      #
+      # @return [Array<Item>]
+      def items
+        addons.map { |addon| item_for(addon) }
+      end
+
+      # Returns the Item matching with given id, if any
+      #
+      # @param id [String]
+      # @return [Item, nil]
+      def find_item(id)
+        addon = find_addon(id)
+
+        return unless addon
+
+        item_for(addon)
+      end
+
+      # Toggles the selection status for the addon represented by given item
+      #
+      # @param item [Item]
+      def toggle_item_selection(item)
+        addon = find_addon(item.id)
+        addon && addon.toggle_selected
+      end
+
+      # Returns a collection holding selected addons
+      #
+      # @see Addon.selected
+      #
+      # @return [Array<Addon>]
+      def selected_items
+        selected_addons = Addon.selected
+
+        log.info "Selected addons: #{selected_addons.map(&:name)}"
+
+        selected_addons
+      end
+
+      # Restores selected addons to the initial selection
+      #
+      # @see @initial_addons_selection
+      def restore_selection
+        Addon.selected.replace(initial_addons_selection)
+      end
+
+      # Returns options for the master widget
+      #
+      # @see Widgets::MasterDetailSelector
+      #
+      # @return [Hash]
+      def master_options
+        {}
+      end
+
+      # Returns options for the details widget
+      #
+      # @see Widgets::MasterDetailSelector
+      #
+      # @return [Hash]
+      def detail_options
+        {
+          placeholder: _("Select an extension or a module to show details here")
+        }
+      end
+
+      # Whether {Widgets::MasterDetailSelector} should include a filter option
+      #
+      # @see Widgets::MasterDetailSelector
+      # @see #development_addons
+      #
+      # @return [Boolean] true if there is any development; false otherwise
+      def include_filter?
+        !development_addons.empty?
+      end
+
+      # Whether {Widgets::MasterDetailSelector} should include a filter option
+      #
+      # @see Widgets::MasterDetailSelector
+      # @see #development_addons
+      #
+      # @return [Boolean] true if there is any development; false otherwise
+      def filter_label
+        _("&Hide Development Versions")
+      end
+
+      # Returns the current value for the filter
+      #
+      # @see .filtering_unreleased
+      #
+      # @return [Boolean] true if filter is active; false otherwise
+      def filter
+        self.class.filtering_unreleased
+      end
+
+      # Updates the filter value
+      #
+      # @param enabled [Boolean] the new value for the filter
+      def filter=(enabled)
+        self.class.filtering_unreleased = enabled
+      end
+
+      # FIXME: still relevant?
```

Comment:
I'd like to know if this piece of code still being relevant. I have ever seen that screen with two columns to introduce registration codes.

Could you please take a look and give me a clue?

Thanks!

Reply:
Thanks for the info @lslezak 

Maybe I'm wrong, but I'd say that I saw that _registration code request_ in a single dialog per addon. That's why I asked if the code still relevant.

Of course, keeping the code there does not hurt.